### ISSUE TITLE
Modified "Comment / Uncomment Selection" shortcut does not work

### ISSUE THREAD

**Issue by cbrnr** [rstudio/rstudio#12871]
### System details

    RStudio Edition : Desktop
    RStudio Version : 2022.12.0 Build 353
    OS Version      : macOS Ventura 13.2.1
    R Version       : 4.2.3

### Steps to reproduce the problem
1. Select "Tools" -> "Modify Keyboard Shortcuts..."
2. Type "comment" in the Filter textbox
3. Change the shortcut for "Comment / Uncomment Selection" to "Cmd + #"
4. Click "Apply"
5. In an R script, try to comment/uncomment a line with "Cmd + #" – it doesn't work

### Describe the problem in detail
The default shortcut "Shift + Cmd + C" works.

### Describe the behavior you expected
I expected that the modified keyboard shortcut "Cmd + #" works (i.e. comments/uncomments lines of code).

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@cbrnr Thank you for raising this! Are you by chance using a QWERTZ or QWERTZU keyboard? I ask because in these, the `#` is a main key. However, in a US QWERTY keyboard, `#` is `Shift-3`, so `Cmd-#` is `Cmd-Shift-3` which is reserved on MacOS for screenshots—this could be causing the problem.

**Comment by cbrnr** [comment]
Yes, I'm using a QUERTZ keyboard. I am pretty sure that shortcut was working in an older RStudio version (probably before the switch to the Electron-based app).

**Comment by ronblum** [comment]
@cbrnr Thanks for the info! We've run into this kind of issue before. I'll mark it for review.

**Comment by cbrnr** [comment]
Is there anything I can do to help move this along? This is one of the most important keyboard shortcuts for me...
